runtime/src/main/java/org/corfudb/runtime/exceptions/unrecoverable/UnrecoverableCorfuInterruptedError.java

```diff
@@ -14,6 +14,8 @@
      */
     public UnrecoverableCorfuInterruptedError(@Nonnull InterruptedException cause) {
         super(cause);
+        // Restore the interrupted status.
```

Comment:
Exceptions are also part of api, can you please add comments to UnrecoverableCorfuInterruptedError about its behavior of setting the interrupt flag?